packages/opentelemetry-core/src/trace/Span.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { randomSpanId, randomTraceId } from '../platform';
+import { isValid } from './spancontext-utils';
+import { performance } from 'perf_hooks';
+
+/**
+ * This class represents a span.
+ */
+export class Span implements types.Span {
+  private _traceId: string;
+  private _spanId: string;
+  private _parentId?: string;
+  private _traceState?: types.TraceState;
+  private _kind: types.SpanKind;
+  private _attributes: types.Attributes = {};
+  private _links: types.Link[] = [];
+  private _events: types.Event[] = [];
+  private _status: types.Status = types.DEFAULT_STATUS_OK;
+  private _ended = false;
+  private _startTime: number;
+  private _duration = 0;
+
+  /** Constructs a new Span instance. */
+  constructor(
+    readonly _parentTracer: types.Tracer,
+    private _spanName: string,
+    readonly _options: types.SpanOptions
+  ) {
+    this._kind = _options.kind || types.SpanKind.INTERNAL;
+    this._spanId = randomSpanId();
+    const spanContext = this._getSpanContext(_options.parent);
+    if (spanContext && isValid(spanContext)) {
+      // New child span.
+      this._traceId = spanContext.traceId;
+      this._parentId = spanContext.spanId;
+      this._traceState = spanContext.traceState;
+    } else {
+      // This is a root span so no remote or local parent.
+      this._traceId = randomTraceId();
+    }
+    this._startTime = _options.startTime || performance.now();
+  }
+
+  tracer(): types.Tracer {
+    return this._parentTracer;
+  }
+
+  context(): types.SpanContext {
+    const spanContext: types.SpanContext = {
```

Comment:
Why is a new `SpanContext` created every time this is called? Why not simply create it in the constructor and move these properties on `SpanContext` directly?

Reply:
Sure, made the change.